### PR TITLE
Quite some stuff

### PR DESCRIPTION
**Pull request by ThexXTURBOXx**
- General cleanup
 - Fix some very obvious issues
 - Mark pack format 7 as supported (will definetly need some additional internal changes, but for the basic stuff, it works)
 - Update dependencies

### CODE REVIEW

File: src/main/java/org/geysermc/packconverter/api/utils/ImageUtils.java [hunk 1]
```diff
@@ -284,7 +283,7 @@ public static boolean isEmptyArea(BufferedImage img, int subX, int subY, int wid
         BufferedImage subImage = img.getSubimage(subX, subY, width, height);
         for (int x = 0; x < subImage.getWidth(); x++) {
             for (int y = 0; y < subImage.getHeight(); y++) {
-                if (!(new Color(subImage.getRGB(x, y), true).equals(Color.TRANSLUCENT))) {
+                if (new Color(subImage.getRGB(x, y), true).getTransparency() != Color.TRANSLUCENT) {
```
Comment (rtm516) [inline src/main/java/org/geysermc/packconverter/api/utils/ImageUtils.java:286]: Don't think this will have the same effect, needs checking.

Comment (ThexXTURBOXx) [reply src/main/java/org/geysermc/packconverter/api/utils/ImageUtils.java:286]: In fact, the old `equals` check will never be `true` (and thus, its negation will never be `false`) as `new Color(subImage.getRGB(x, y), true)` is type `Color` and `Color.TRANSLUCENT` is type `int`.

I believe, this new check is the one you originally wanted to have there. If you don't want any new behavior, just remove the check entirely.

**Comment by ThexXTURBOXx** [comment]
Rebased on `master`. Ready for review again.

File: src/main/java/org/geysermc/packconverter/api/utils/ImageUtils.java [hunk 2]
```diff
@@ -256,16 +255,17 @@ private static BufferedImage toBufferedImage(Image img) {
...
      */
     public static BufferedImage rotate(BufferedImage img, int angle) {
+		final int type = img.getType();
         final double rads = Math.toRadians(-angle);
         final double sin = Math.abs(Math.sin(rads));
         final double cos = Math.abs(Math.cos(rads));
         final int w = (int) Math.floor(img.getWidth() * cos + img.getHeight() * sin);
         final int h = (int) Math.floor(img.getHeight() * cos + img.getWidth() * sin);
-        final BufferedImage rotatedImage = new BufferedImage(w, h, img.getType());
+        final BufferedImage rotatedImage = new BufferedImage(w, h, type == 0 ? 5 : type);
```
Comment (rtm516) [inline src/main/java/org/geysermc/packconverter/api/utils/ImageUtils.java:264]: Why this change?

File: src/main/java/org/geysermc/packconverter/api/utils/ImageUtils.java [hunk 3]
```diff
@@ -256,16 +255,17 @@ private static BufferedImage toBufferedImage(Image img) {
      * @return Rotated image
      */
     public static BufferedImage rotate(BufferedImage img, int angle) {
+		final int type = img.getType();
```
Comment (rtm516) [inline src/main/java/org/geysermc/packconverter/api/utils/ImageUtils.java:258]:
```suggestion
	final int type = img.getType();
```

File: src/main/java/org/geysermc/packconverter/api/utils/ImageUtils.java (hunk 2 above)
Comment (ThexXTURBOXx) [reply src/main/java/org/geysermc/packconverter/api/utils/ImageUtils.java:264]: It's a workaround for if something goes wrong.
There's not really a better solution for that afaik.
Problem occurs for example when converting Xmas BDCraft 1.17 64x.

**Comment by Redned235** [comment]
Sorry for the late response here and thanks for the PR! This project is currently in the process of a rewrite after not being actively maintained for a few years (changes here: https://github.com/GeyserMC/PackConverter/tree/feature/refactor).

Given the significant changes, unfortunately these will no longer merge with that branch and a handful of the changes from this PR aren't entirely necessary anymore. I'm going to close this PR, but if there are any changes you deem relevant in the rewrite, feel free to open a new pull request onto that branch. Thanks!
